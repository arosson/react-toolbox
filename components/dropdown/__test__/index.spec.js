import React from 'react';
import { shallow } from 'enzyme';
import { Dropdown } from '../Dropdown';
import theme from '../theme.css';

const countries = [
  { value: 'EN-gb', label: 'England', img: 'http://' },
  { value: 'ES-es', label: 'Spain', img: 'http://' },
  { value: 'TH-th', label: 'Thailand', img: 'http://', disabled: true },
  { value: 'EN-en', label: 'USA', img: 'http://' },
  { value: 'FR-fr', label: 'France', img: 'http://' },
];


describe('Dropdown', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Dropdown theme={theme} source={countries} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
