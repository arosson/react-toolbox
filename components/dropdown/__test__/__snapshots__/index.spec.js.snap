// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onChange={[Function]}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={null}
    themeNamespace="input"
    type={null}
    value=""
  />
  <ul
    className="values"
    onKeyDown={[Function]}
  >
    <li
      className=""
      key="0"
      onFocus={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      England
    </li>
    <li
      className=""
      key="1"
      onFocus={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      Spain
    </li>
    <li
      className="disabled"
      key="2"
      onFocus={[Function]}
      tabIndex={-1}
    >
      Thailand
    </li>
    <li
      className=""
      key="3"
      onFocus={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      USA
    </li>
    <li
      className=""
      key="4"
      onFocus={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      France
    </li>
  </ul>
</div>
`;
